### PR TITLE
[Docs] Update and Align Metric Reference

### PR DESCRIPTION
**Pull request by alexfornuto**
This is a recreation of #14729, which was pointed at a file path that no longer exists, and git was being... git.

Original description:

<details><summary>OG Description</summary>
On https://goteleport.com/docs/setup/reference/metrics/ , we had a customer report that:

* the metric name in the docs ( `heartbeat_connections_missed_total`) doesn't match the actual metric name - `heartbeats_missed_total`.

This PR corrects this.  

Also, I curled the teleport metrics endpoint on a controller node and noticed a bunch of other metrics that are missing (or named differently) in the docs as well:

```
backend_requests Number of write requests to the backend [Also fixed in this PR]
bpf_lost_command_events Number of lost command events. [missing, do we want to place this in a new BPF section?]
bpf_lost_disk_events Number of lost disk events. [missing, do we want to place this in a new BPF section?]
bpf_lost_network_events Number of lost network events. [missing, do we want to place this in a new BPF section?]
firestore_events_backend_write_requests Number of write requests to firestore events [missing]
firestore_events_backend_write_seconds Latency for firestore events write operations [missing]
proxy_ssh_sessions_total Number of active sessions through this proxy [missing]
```

Feel free to let me know where these should be placed and I can add them.
</details>

TL;DR: Some metrics reference lines were incorrect or missing.

To whoever reviews this from @gravitational/employees-development, please add additional backport labels as appropriate. 
**EDIT:** Apologies for the mass ping.

**Note for reviewers:** I formatted the tables to be easier to parse while editing. Formatting is done in one commit and actual changes in another, so you can look at the second commit to see just the content changes.

All credit to @ulysseskan as the original author.

### CODE REVIEW

**Review by gozer — APPROVED** [review]

**Review by strideynet — APPROVED** [review]

File: docs/pages/reference/metrics.mdx [hunk 1]
```diff
@@ -83,75 +83,86 @@ Teleport Cloud does not expose monitoring endpoints for the Auth Service and Pro
...
+| `etcd_backend_read_seconds`                     | histogram   | etcd                | Latency for etcd read operations.                                                                  |
+| `etcd_backend_tx_requests`                      | counter     | etcd                | Number of transaction requests to the database.                                                    |
+| `etcd_backend_tx_seconds`                       | histogram   | etcd                | Latency for etcd transaction operations.                                                           |
+| `etcd_backend_write_requests`                   | counter     | etcd                | Number of write requests to the database.                                                          |
+| `etcd_backend_write_seconds`                    | histogram   | etcd                | Latency for etcd write operations.                                                                 |
+| `firestore_events_backend_batch_read_requests`  | counter     | GCP Cloud Firestore | Number of batch read requests to Cloud Firestore events.                                           |
```
Comment (ptgott) [inline docs/pages/reference/metrics.mdx:117]: Should "GCP Cloud Firestore" (or just "Cloud Firestore") be "Google Cloud Firestore" here and below?

File: docs/pages/reference/metrics.mdx [hunk 2]
```diff
@@ -83,75 +83,86 @@ Teleport Cloud does not expose monitoring endpoints for the Auth Service and Pro
...
+| `grpc_client_msg_received_total`          | counter | Teleport Proxy | Total number of RPC stream messages received on the client.                                                                     |
+| `grpc_client_msg_sent_total`              | counter | Teleport Proxy | Total number of gRPC stream messages sent by the client.                                                                        |
+| `proxy_connection_limit_exceeded_total`   | counter | Teleport Proxy | Number of connections that exceeded the proxy connection limit.                                                                 |
+| `proxy_ssh_sessions_total`                | gauge   | Teleport Proxy | Number of active sessions through this proxy.                                                                                   |
+| `proxy_missing_ssh_tunnels`               | gauge   | Teleport Proxy | Number of missing SSH tunnels. Used to debug if nodes have discovered all proxies.                                              |
```
Comment (ptgott) [inline docs/pages/reference/metrics.mdx:163]:
```suggestion
| `proxy_missing_ssh_tunnels`               | gauge   | Teleport Proxy | Number of missing SSH tunnels. Used to debug if Nodes have discovered all Proxy Service instances.                                              |
```

**Review by ptgott — APPROVED** [review]
Approved with suggestions. These apply to the copy as it was before the PR, but I thought we could use this as an opportunity to improve this page a bit.

File: docs/pages/reference/metrics.mdx (hunk 1 above)
Comment (alexfornuto) [reply docs/pages/reference/metrics.mdx:117]: Googling for "Google Cloud Firestore" and "GCP Cloud Firestore" both bring up the same results, while just "Cloud Firestore" bring up a less relevant first hit. Between "Google" and "GCP", I think we can trust the audience of this page to know that acronym, and tables like this are the place to be concise.
